docs/features/incremental-generators.md

```diff
@@ -428,51 +1031,122 @@ In this way, only changes that are consequential flow through the pipeline, and
+For a user provided result to be comparable across iterations, there needs to be
+some concept of equivalence. By default the host will use `EqualityComparer<T>`
+to determine equivalence. There are obviously times where this is insufficient,
+and there exists an extension method that allows the author to supply a comparer
+that should be used when comparing values for the given transformation:
 
 ```csharp
-public static partial class IncrementalValueSourceExtensions
+public static partial class IncrementalValueProviderExtensions
 {
-    public static IncrementalValueSource<T> WithComparer(IEqualityComparer<T> comparer) => ...
+        public static IncrementalValueProvider<TSource> WithComparer<TSource>(this IncrementalValueProvider<TSource> source, IEqualityComparer<TSource> comparer);
+
+        public static IncrementalValuesProvider<TSource> WithComparer<TSource>(this IncrementalValuesProvider<TSource> source, IEqualityComparer<TSource> comparer);
 }
 ```
 
-Allowing the user to specify a given comparer.
+Allowing the generator author to specify a given comparer.
 
 ```csharp
-var withComparer = context.Sources.AdditionalTexts
-                                  .Transform(t => t.Path)
-                                  .WithComparer(myComparer);
+var withComparer = context.AdditionalTextsProvider
+                          .Select(t => t.Path)
+                          .WithComparer(myComparer);
 ```
 
-Note that the comparer is on a per-transformation basis, meaning an author can specify different comparers for different parts of the pipeline. 
+Note that the comparer is on a per-transformation basis, meaning an author can
+specify different comparers for different parts of the pipeline.
 
 ```csharp
-var transform = context.Sources.AdditionalTexts.Transform(t => t.Path);
+var select = context.AdditionalTextsProvider.Select(t => t.Path);
 
-var noCompareTransform = transform.Transform(...);
-var compareTransform = transform.WithComparer(myComparer).Transform(...);
+var noCompareSelect = select.Select(...);
+var compareSelect = select.WithComparer(myComparer).Select(...);
 ```
 
-The same transform node can have no comparer when acting as input to one step,
-and still provide one when acting as input to a different step.
+The same select node can have no comparer when acting as input to one transformation,
+and still provide one when acting as input to a different transformation.
+
+The host will only invoke the given comparer when the item it is derived from
+has been modified. When the input value is new or being removed, or the input
+transformation was determined to be cached (possibly by a provided comparer) the
+given comparer is not considered.
+
+### Authoring a cache friendly generator
```

Comment:
This section probably wants to be expanded into another doc such as the cookbook, but I wanted to get a least a few 'best practices' down that we can point to for now until we have the expanded versions.